### PR TITLE
Add a Vagrant image for use on OSX and Windows

### PR DESCRIPTION
**Pull request by stieg**
Adds the ability to build firmware on any machine using a vagrant image that will set itself up automatically.

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+# you're doing.
+Vagrant.configure(2) do |config|
+  # The most common configuration options are documented and commented below.
+  # For a complete reference, please see the online documentation at
+  # https://docs.vagrantup.com.
+
+  # Every Vagrant development environment requires a box. You can search for
+  # boxes at https://atlas.hashicorp.com/search.
+  config.vm.box = 'holms/debian-jessie-amd64'
+  config.vm.box_url = "https://github.com/holms/vagrant-jessie-box/releases/download/Jessie-v0.1/Debian-jessie-amd64-netboot.box"
```
Comment (brentpicasso) [inline Vagrantfile]: How was this box selected? is there an official ultra-generic box we can use?

Comment (stieg) [inline Vagrantfile]: I pretty much went to their site and found one that I wanted to play with.  I could change it to any Debian derivative (ie Ubuntu).

Comment (brentpicasso) [inline Vagrantfile]: Would prefer a generic box that would likely never go away. Plus gives warm fuzzies to people using this vagrant script - if they see a box from an official source, they will have confidence in using it.

Comment (stieg) [inline Vagrantfile]: To be honest this VM is on their ATLAS site.  I have a really hard time telling what is legit with Vagrant and what isn't.  If you have a recommendation I am all ears.  Could probably switch it over to https://atlas.hashicorp.com/ubuntu/boxes/trusty64 if you think that is better.  We can also host a box like that on s3 if needed to guarantee that it works.

**Pull request merged**